mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+
+    flavor_conf = _get_flavor_configuration_from_uri(model_uri, FLAVOR_NAME, _logger)
+    local_mlflow_model_path = _download_artifact_from_uri(
+        artifact_uri=model_uri, output_path=dst_path
+    )
+    _add_code_from_conf_to_system_path(local_mlflow_model_path, flavor_conf)
+
+    if _should_use_mlflowdbfs(model_uri):
+        from pyspark.ml.pipeline import PipelineModel
+
+        mlflowdbfs_path = _mlflowdbfs_path(
+            DatabricksArtifactRepository._extract_run_id(model_uri), artifact_path
+        )
+        with databricks_utils.MlflowCredentialContext(
+                get_databricks_profile_uri_from_artifact_uri(root_uri)
+        ):
+            return PipelineModel.load(mlflowdbfs_path)
+
+    sparkml_model_uri = append_to_uri_path(model_uri, flavor_conf["model_data"])
+    local_sparkml_model_path = os.path.join(local_mlflow_model_path, flavor_conf["model_data"])
+    return _load_model(
+        model_uri=sparkml_model_uri,
+        dfs_tmpdir_base=dfs_tmpdir,
+        local_model_path=local_sparkml_model_path,
+    )
+
+
+def _load_pyfunc(path, spark=None):
+    """
+    Load PyFunc implementation. Called by ``pyfunc.load_model``.
+    :param path: Local filesystem path to the MLflow Model with the ``johnsnowlabs`` flavor.
+    :param spark: Optionally pass spark context when using pyfunc as UDF. required, because we cannot fetch the Sparkcontext inside of the Workernode which executes the UDF
+    :return:
+    """
+    return _PyFuncModelWrapper(_load_model(model_uri=path), spark if spark else get_or_create_sparksession(path))
+
+
+def get_or_create_sparksession(model_path=None):
+    """
+    1. Check if SparkSession running and get it
+    2. If none exists, create a new one using jars in model_path
+    3. If model_path not defined, rely on nlp.start() to create a new
+    one using johnsnowlabs Jar resolution method
+    See https://nlp.johnsnowlabs.com/docs/en/jsl/start-a-sparksession
+    and https://nlp.johnsnowlabs.com/docs/en/jsl/install_advanced
+    :param model_path:
+    :return:
+    """
+    validate_env_vars()
+    from mlflow.utils._spark_utils import _get_active_spark_session
+    spark = _get_active_spark_session()
+    if spark is None:
+        spark_conf = {}
+        spark_conf["spark.python.worker.reuse"] = "true"
+        # disable task retry (i.e. make it fast fail)
+        spark_conf["spark.task.maxFailures"] = "1"
+        # Disable simplifying traceback from Python UDFs
+        spark_conf["spark.sql.execution.pyspark.udf.simplifiedTraceback.enabled"] = "false"
+        # Show jvm side stack trace.
+        spark_conf["spark.sql.pyspark.jvmStacktrace.enabled"] = "true"
+        spark_conf["spark.jars.excludes"] = "net.sourceforge.f2j:arpack_combined_all"
+        spark_conf["spark.driver.extraJavaOptions"] = "-Dio.netty.tryReflectionSetAccessible=true"
+        spark_conf["spark.executor.extraJavaOptions"] = "-Dio.netty.tryReflectionSetAccessible=true"
```

Comment:
```suggestion
        spark_conf = {
            "spark.python.worker.reuse": "true",
            ...
        }
```

Reply:
thank you, I added these configs from [test_mleap_model_export.py](https://github.com/mlflow/mlflow/blob/bcacb9fb1381f3e0dcb93fef535c2da2d56e8908/tests/mleap/test_mleap_model_export.py#L47-L55) and [test_spark.py](https://github.com/mlflow/mlflow/blob/bcacb9fb1381f3e0dcb93fef535c2da2d56e8908/tests/pyfunc/test_spark.py#LL86C2-L104C14)  but seems like they are irrelevant, dropping them